content/components/dialog.mdx

```diff
@@ -447,6 +461,45 @@ This is because some browser and screen reader combinations may ignore the live
 
 If you need to use a live region while a dialog is active, place the live region within the dialog, rather than outside of it.
 
+## Built-in accessibility features
+
+The Rails version of the component generates a `<dialog>` element with `aria-modal="true"`.
+
+The React version of the component generates a `<div>` element with `role="dialog"` and `aria-modal="true"`.
+
+When provided, the `title` attribute/property for the component is rendered as an `<h1>` heading, and is explicitly referenced by the dialog using `aria-labelledby` to provide an accessible name for the dialog.
+
+By default, the generated dialog includes an "X" graphical close control, which receives focus automatically when the dialog is opened.
+
+In the Rails version of the component, you can use the `autofocus` property on interactive elements in the dialog's content to move the initial focus directly to a specific control. However, you should carefully consider when this approach is necessary – in general, it is best to keep dialogs consistent and let focus go to the "X" close control.
+
+Focus is maintained inside the dialog. The `aria-modal="true"` on the dialog hides the underlying page from screen readers.
+
+When the dialog is closed, focus is automatically moved back to the last element in the underlying page that had focus - generally, the control the triggered the dialog.
+
+## Implementation requirements
+
+Make sure to provide a sufficiently descriptive `title` for the modal dialog. The title should concisely describe the contents or purpose of the modal.
```

Comment:
It might be worth mentioning that the `title` is rendered in the "heading" area of the dialog - it's not the `title` attribute of anything.